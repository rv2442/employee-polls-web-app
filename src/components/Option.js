import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";

const Option = ({
	textOption,
	option,
	handleOption,
	answeredOption,
	percentageOption,
	numberOfVotesOption,
}) => {
	return (
		<Card style={{ width: "18rem" }}>
			<Card.Body>
				<Card.Text>{textOption}</Card.Text>
				<Button
					style={{ width: "auto" }}
					variant="primary"
					onClick={handleOption}
					value={option}
					disabled={answeredOption !== "NOT_ANSWERED"}
				>
					Click
				</Button>
				{answeredOption === option ? (
					<Card.Text style={{ color: "green" }}> ✔️ Option selected.</Card.Text>
				) : null}
				{answeredOption !== "NOT_ANSWERED" ? (
					<Card.Text style={{ color: "blue" }}>
						{" "}
						People voted: {numberOfVotesOption}
					</Card.Text>
				) : null}
				{answeredOption !== "NOT_ANSWERED" ? (
					<Card.Text style={{ color: "blue" }}>
						{" "}
						Percentage voted: {percentageOption} %
					</Card.Text>
				) : null}
			</Card.Body>
		</Card>
	);
};
export default Option;
