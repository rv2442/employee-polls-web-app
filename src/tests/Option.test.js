import * as React from "react";
import { render } from "@testing-library/react";
import Option from "../components/Option";

describe("Option", () => {
	it("matches the snapshot when al the parameters are passed and the NOT_ANSWERED is selected", () => {
		var component = render(
			<Option
				textOption={"Play futbol"}
				option={"optionOne"}
				answeredOption={"NOT_ANSWERED"}
				percentageOption={50}
				numberOfVotesOption={5}
			/>
		);
		expect(component).toMatchSnapshot();
	});

	it("matches the snapshot when al the parameters are passed and the optionOne is selected", () => {
		var component = render(
			<Option
				textOption={"Play futbol"}
				option={"optionOne"}
				answeredOption={"optionOne"}
				percentageOption={50}
				numberOfVotesOption={5}
			/>
		);
		expect(component).toMatchSnapshot();
	});
});
