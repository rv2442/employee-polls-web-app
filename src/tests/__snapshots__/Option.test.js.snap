// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option matches the snapshot when al the parameters are passed and the NOT_ANSWERED is selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        style="width: 18rem;"
      >
        <div
          class="card-body"
        >
          <p
            class="card-text"
          >
            Play futbol
          </p>
          <button
            class="btn btn-primary"
            style="width: auto;"
            type="button"
            value="optionOne"
          >
            Click
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      style="width: 18rem;"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          Play futbol
        </p>
        <button
          class="btn btn-primary"
          style="width: auto;"
          type="button"
          value="optionOne"
        >
          Click
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Option matches the snapshot when al the parameters are passed and the optionOne is selected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
        style="width: 18rem;"
      >
        <div
          class="card-body"
        >
          <p
            class="card-text"
          >
            Play futbol
          </p>
          <button
            class="btn btn-primary"
            disabled=""
            style="width: auto;"
            type="button"
            value="optionOne"
          >
            Click
          </button>
          <p
            class="card-text"
            style="color: green;"
          >
             ✔️ Option selected.
          </p>
          <p
            class="card-text"
            style="color: blue;"
          >
             
            People voted: 
            5
          </p>
          <p
            class="card-text"
            style="color: blue;"
          >
             
            Percentage voted: 
            50
             %
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
      style="width: 18rem;"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          Play futbol
        </p>
        <button
          class="btn btn-primary"
          disabled=""
          style="width: auto;"
          type="button"
          value="optionOne"
        >
          Click
        </button>
        <p
          class="card-text"
          style="color: green;"
        >
           ✔️ Option selected.
        </p>
        <p
          class="card-text"
          style="color: blue;"
        >
           
          People voted: 
          5
        </p>
        <p
          class="card-text"
          style="color: blue;"
        >
           
          Percentage voted: 
          50
           %
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
